cli/slsa-verifier/main_test.go

```diff
@@ -491,34 +495,87 @@ func Test_runVerifyArtifactPath(t *testing.T) {
 				artifactPath := filepath.Clean(filepath.Join(TEST_DIR, v, tt.artifact))
 				provenancePath := fmt.Sprintf("%s.intoto.jsonl", artifactPath)
 
-				cmd := verify.VerifyArtifactCommand{
-					ProvenancePath:      provenancePath,
-					SourceURI:           tt.source,
-					SourceBranch:        tt.pbranch,
-					BuilderID:           tt.pbuilderID,
-					SourceTag:           tt.ptag,
-					SourceVersionTag:    tt.pversiontag,
-					BuildWorkflowInputs: tt.inputs,
+				// TODO(#258): invalid builder ref.
+				sv := path.Base(v)
+				// For each test, we run 4 sub-tests:
+				// 	1. With the the full builderID including the semver in short form.
+				//	2. With the the full builderID including the semver in long form.
+				//	3. With only the name of the builder.
+				//	4. With no builder ID.
+				var builder string
+				// Select the right builder based on directory structure.
+				parts := strings.Split(v, "/")
+				name := parts[0]
+				version := ""
+				if len(parts) > 1 {
+					version = parts[1]
+				}
+				switch {
+				case strings.HasSuffix(name, "_go"):
+					builder = goBuilder
+				case strings.HasSuffix(name, "_generic"):
+					builder = genericBuilder
+				default:
+					builder = genericBuilder
 				}
 
-				outBuilderID, err := cmd.Exec(context.Background(), []string{artifactPath})
+				// Default builders to test.
+				builderIDs := []*string{
+					pString(builder),
+					nil,
+				}
 
-				if !errCmp(err, tt.err) {
-					t.Errorf(cmp.Diff(err, tt.err, cmpopts.EquateErrors()))
+				// We only add the tags to tests for versions >= 1,
+				// because we generated them with a builder at `@main`
+				// before GA. Add the tests for tag verification.
+				if version != "" && semver.Compare(version, "v1.0.0") > 0 {
```

Comment:
Note: we should update the no-version tests to be from a tag, and append the `builder@tag` for verification